packages/server/src/integrations/dynamodb.js

```diff
@@ -0,0 +1,72 @@
+const AWS = require("aws-sdk")
+const { FIELD_TYPES, QUERY_TYPES } = require("./Integration")
+
+const SCHEMA = {
+  datasource: {
+    region: {
+      type: FIELD_TYPES.STRING,
+      required: true,
+      default: "us-east-1",
+    },
+    accessKeyId: {
+      type: FIELD_TYPES.PASSWORD,
+      required: true,
+    },
+    secretKey: {
+      type: FIELD_TYPES.PASSWORD,
+      required: true,
+    },
+  },
+  query: {
```

Comment:
Sorry I'm not starting a code review, was just looking through all the examples of the integrations that exist, ran into the DynamoDB one (of course it would catch my eye haha) and was wondering two things about it:
1. What way do we handle when the schema isn't "complete" - by that I mean the integrator hasn't added say create, update or delete functionality, do we just disable those functions in the builder UI?
2. I see here the `ExpressionAttributeNames` and `ExpressionAttributeValues` have been set as type string, does the UI for building queries like this allow building arbitrary depth objects, as in the `ExpressionAttributeNames` is usually an object, like `{"#prop1": "property-name"}` - is the type set to string because the UI builds JSON strings rather than pure objects? Just thinking if this is the case then maybe we should have a custom type, like the type for JSON schema standards is a string, but we have a custom property like `customType` which specifies that its really JSON for validation reasons. This is something we do with the automation schema.

Reply:
If a CRUD method is missing from the schema, it just won't be choosable from the UI. Some integrations don't really make sense to have create/update methods (like s3, for example).

This one hasn't been updated yet - that string version is the original schema when we did lab day, but for `ExpressionAttributeNames` and `ExpressionAttributeValues` I think we should just have a JSON type, and allow people to just write JSON in there. We could of course abstract this a little into keys and values, but tbh I think it's actually easier to just write JSON than it is to set explicit types or coerce values from the UI. 

Edit: Just pushed the updated schema. The `customisable` key allows you to append JSON to your query on top of some required fields. This should allow people to interact directly with the dynamo API.

Reply:
Sounds good to me, that's an ideal way to handle it, I think just allowing JSON for these type of things makes sense, theres no way we could build an actual UI to build every type of Dynamo query, whereas allowing JSON for each of the attributes is a nice quick method to allow any sort of query.

If we were to try and build custom UIs for every integration type (e.g. allow specifying specific key values which pair up with the `KeyConditionExpression`) we'd have to do the same for every integration and it would get out of hand far too quickly. Allowing JSON is an easy solution and is just as good as you'd get from actual API. 

One nice thing we could do, is as part of the schema/integration information we could provide a link to the docs, make it standard practice to link to the docs that should make it clear how to interact with the API? Then have like `Find more information about the DynamoDB API here`.

Reply:
I thought about this actually - I think retool do it. It's a nice touch to link them to the API docs. I'll add a field in the schema that points to the docs and allows them to visit from the builder.

Reply:
@mike12345567 I tried out the JSON schema configuration - while it makes things more consistent with automations it makes things quite different from tables, introducing quite a bit of conditional logic into the builder frontend. Keeping it as is for now, but will introduce yup/joi validation on integration schemas in future to make sure everything is correct.